src/components/inviteWelcome/InviteWelcome.js

```diff
@@ -48,109 +49,132 @@ const steps = {
 
 const InviteWelcome = ({ styles, screenProps, navigation }) => {
   const [step, setStep] = useState(1)
+  const [show, setShow] = useState(undefined)
+
   const { navigate } = navigation
 
+  const skipIntro = useCallback(() => {
+    fireEvent(INVITEWELCOME_SKIPPED, { step })
+    return done()
+  }, [navigate])
   const goToSignUp = useCallback(() => {
     fireEvent(INVITEWELCOME_SKIPPED, { step })
 
     return navigate('Welcome')
   }, [navigate, step])
 
+  const done = () => {
+    AsyncStorage.setItem('isFirstTime', false)
```

Comment:
Please use a separate hook for that flag (as i did with GoodMarket)
1. there're some ready to use solutions like this one https://github.com/react-native-hooks/async-storage/blob/master/src/index.js
- you could just copy it's code and modify it to use our `src/lib/utils/asyncStorage.js` 
- you could write then your own hook _using_ useAsyncStorage which implements the logic you need (check flag, return currrent viewed/not viewed status, update status immediately on 'first view' etc)
2. or you could just write a hook which contains only the logic you need and working with AsyncStorage directly

Reply:
before suggesting HOW
explain WHY
i dont see a very good reason to use a hook for a very simple library like asyncstorage.
hooks makes sense if you want to encapsulate complicated logic

Reply:
hooks makes sense if you want to incapsulate some logic in a single place
i see then you're getting/setting that flag in some places. what if we'll change the localStorage's key for it ?
keeping it in a hook will simplify this feature support. and hook will be very simple you won't spend a lot of time to write it